Sveriges radio.js

```diff
@@ -0,0 +1,709 @@
+	if (detectWeb(doc, url) == "multiple") {
+		Zotero.selectItems(getSearchResults(doc, false), function (items) {
+			if (!items) {
+				return true;
+			}
+			var articles = [];
+			for (var i in items) {
+				articles.push(i);
+			}
+			ZU.processDocuments(articles, scrape);
+		});
+	} else {
+		scrape(doc, url);
+	}
+}
+
+
+function scrape(doc, url) {
+	var translator = Zotero.loadTranslator('web');
+	// Embedded Metadata
+	translator.setTranslator('951c027d-74ac-47d4-a107-9c3069ab7b48');
+	// translator.setDocument(doc);
+	
+	translator.setHandler('itemDone', function (obj, item) {
+		item.publicationTitle = "Sveriges Radio";
+
+		item.creators = [];
+		var nameNodes = ZU.xpath(doc, '//p[@class="byline"]/text()');
+		for(var node of nameNodes) {
+			var nameString = node.wholeText;
+			if(nameString.trim() === "") {
+				// Empty text node are added between authors; we want to skip
+				// those.
+				continue;
+			}
+			var names = nameString.split(" ");
+			var firstName = names[0];
+			var lastName = names[1];
+			if(lastName === undefined) {
+				lastName = "";
+			}
+			var author = {
+				// Assume only one first name.
+				firstName: firstName,
+				lastName: lastName,
+				creatorType: "author"
+			};
+			item.creators.push(author);
+		}
+
+		// The title from the meta is in the format:
+		// Australierna säger ja till samkönade äktenskap - Nyheter (Ekot)
+		var titleString =
+			ZU.xpath(doc, '//meta[@name="twitter:title"]')[0]
+			.getAttribute("content");
+		var titleParts = titleString.split(" - ");
+		item.title = titleParts[0];
+		item.section = titleParts[1];
+
+		var dateString = 
+			ZU.xpath(doc, '//meta[@name="displaydate"]')[0]
+			.getAttribute("content");
+		item.date = dateString.replace(/(....)(..)(..)/, "$1-$2-$3");
+
+		item.complete();
+	});
+
+	translator.getTranslatorObject(function(trans) {
+		trans.itemType = "newspaperArticle";
+		// TODO map additional meta tags here, or delete completely
+		trans.addCustomFields({
+			'twitter:description': 'abstractNote'
+		});
+		trans.doWeb(doc, url);
+	});
+}
+/** BEGIN TEST CASES **/
+var testCases = [
+	{
+		"type": "web",
+		"url": "https://sverigesradio.se/sida/artikel.aspx?programid=83&artikel=6821850",
+		"items": [
+			{
+				"itemType": "newspaperArticle",
+				"title": "Australierna säger ja till samkönade äktenskap",
+				"creators": [
+					{
+						"firstName": "Ekot",
+						"lastName": "",
+						"creatorType": "author"
+					}
+				],
+				"date": "2017-11-15",
+				"abstractNote": "I en rådgivande postomröstning i Australien svarade 61,6 procent av medborgarna att de ville se en lag som godkänner samkönade äktenskap.",
+				"language": "sv",
+				"libraryCatalog": "sverigesradio.se",
+				"publicationTitle": "Sveriges Radio",
+				"section": "Nyheter (Ekot)",
+				"url": "http://sverigesradio.se/sida/artikel.aspx?programid=83&artikel=6821850",
+				"attachments": [
+					{
+						"title": "Snapshot"
+					}
+				],
+				"tags": [
```

Comment:
These tags look generic and not helpful. I'd clean them out.

Reply:
I wasn't sure what tags did, so I kept them as they were. As you say, at least some of them are quite general and don't seem to relate to the article. Happy to get rid of them.